feat: add MoonBitTarget command

HCHogan opened the pull request:
Added target switching command `MoonbitTarget` using `moon check --target`.
This allows the user to switch targets on the fly without restarting nvim.

tonyfettes reviewed (commented):
Can you please check again if the name/usage of wasm-gc target is correct?

As on the page of moon check documentation it says wasm-gc instead of wasm_gc is the valid target name.

https://docs.moonbitlang.com/en/latest/toolchain/moon/commands.html#moon-check

HCHogan commented:
> Can you please check again if the name/usage of wasm-gc target is correct?
> 
> As on the page of moon check documentation it says wasm-gc instead of wasm_gc is the valid target name.
> 
> https://docs.moonbitlang.com/en/latest/toolchain/moon/commands.html#moon-check

Thank you for catching that! I corrected it and moved the `setup_moonbit_target()` to a module-level initialization
